src/crimson/os/seastore/cache.cc

```diff
@@ -1788,7 +1788,19 @@ Cache::replay_delta(
         return seastar::make_ready_future<CachedExtentRef>();
       }
     };
-    auto extent_fut = (delta.pversion == 0 ?
+    auto need_read_from_disk = [this, &delta]() -> bool {
+      if (delta.paddr.get_addr_type() == paddr_types_t::SEGMENT ||
+	  get_extent_category(delta.type) != data_category_t::DATA) {
+	return delta.pversion == 0;
+      } else {
+	assert(delta.paddr.get_addr_type() == paddr_types_t::RANDOM_BLOCK);
+	// in partial-overwrite, the version of the initial delta can be non-zero,
+	// so we need to call query_cache unconditionally
```

Comment:
Inplace-rewrite flushes some dirty extents to become clean, so no real modifications happen -- I think this logically means there are no conflicts to other transactions working on the same extents.

Refer to comment https://github.com/ceph/ceph/pull/54525#discussion_r1418489309, looks to me dirty extents are updated at prepare_record time, which is an exclusive and ordered phase for all the transactions. So, if inplace-rewrite reset the version of the dirty extent to 0 and set it to clean at prepare_record time, other transactions only need to tolerate that this can happen implicitly for certain types of extents upon prepare_record, does it make sense?

Reply:
Do you mean that we can allow the condition like 
```
assert(next->version == prev->version + 1 || (prev->version == 0 && next->version > 1));
version = 1;
```
instead of  
`assert(next->version == prev->version + 1)` in `commit_replace_extent()`? 
If so, it makes sense.

Reply:
Yeah, reset the version to 1 if the mutated extent was inplace-rewritten right before this transaction, at prepare-record time.

Let's limit this only to the DATA extents, or better, only to the inplace-rewritten extents if possible.

Let's see whether it works.

Reply:
Done.